### PR TITLE
✨(api) Add "mine" query parameter

### PR DESCRIPTION
**Pull request by Leobouloc**
## Purpose

In link with #288 , we need a mechanism to fetch statements by authority. It has been decided that the `GET /statements` endpoint will implement a `mine` option (not in xAPI specification), which is used to return only the statements for which the "authority" field matches the "agent" field of the current user (stored in credentials since #342). 

With future implementation of permissions, it is expected that the use of `mine` option will be made mandatory for users of the API that do not have full read access. The idea behind this is to make any filtering obvious to the user ("explicit is better than implicit").

## Proposal

Add `mine` by adding fetching user "agent" and querying using a new database query field.

- [x] add option to API
- [x] add backend query for authority
- [x] add test for API that covers `mine` option

### CODE REVIEW

File: src/ralph/api/routers/statements.py [hunk 1]
```diff
@@ -61,11 +63,34 @@
 }
 
 
+def _parse_agent_parameters(agent_obj: dict):
```
Comment (wilbrdt) [inline src/ralph/api/routers/statements.py:66]: We could even LRU-cache this method, WDYT?

File: tests/api/test_auth.py [hunk 2]
```diff
@@ -66,3 +66,5 @@ def test_api_auth_model_serveruserscredentials():
                 ),
             ]
         )
+
+    # TODO: should we verify agent uniqueness ?
```
Comment (wilbrdt) [inline tests/api/test_auth.py:70]:
```suggestion
```

File: tests/api/test_statements_get.py [hunk 3]
```diff
@@ -76,36 +84,137 @@ def create_mock_agent(ifi: str, id_: int, home_page_id=None):
...
+    )
+    assert response.status_code == 200
+    assert response.json() == {"statements": [statements[0]]}
+
+    # Check that invalid parameters returns an error
+    response = client.get(
+        "/xAPI/statements/?mine=BigBoat",
+        headers={"Authorization": f"Basic {credentials_1_bis}"},
+    )
+    assert response.status_code == 422  # TODO: is this the proper code
```
Comment (wilbrdt) [inline tests/api/test_statements_get.py:215]: What would you propose? I think 422 is good

File: tests/fixtures/auth.py [hunk 4]
```diff
@@ -9,17 +9,17 @@
 from ralph.conf import settings
 
 
-def create_user(fs_, user: str, pwd: str, scopes: list, agent: dict):
+def create_user(fs_, username: str, password: str, scopes: list, agent: dict):
     """Create a user using Basic Auth in the (fake) file system.
 
     Args:
         fs_: fixture provided by pyfakefs
         user: username used for auth
```
Comment (wilbrdt) [inline tests/fixtures/auth.py:17]:
```suggestion
        username: username used for auth
```

**Review by wilbrdt — COMMENTED** [review]
Just a few remarks but looks good!

File: src/ralph/api/routers/statements.py (hunk 1 above)
Comment (Leobouloc) [reply src/ralph/api/routers/statements.py:66]: Good idea ! I will add this.

File: src/ralph/conf.py [hunk 5]
```diff
@@ -320,6 +320,7 @@ class Config(BaseSettingsConfig):
         },
     }
     PARSERS: ParserSettings = ParserSettings()
+    QUERY_AGENTS_CACHE_MAX_SIZE: int = 100
```
Comment (wilbrdt) [inline src/ralph/conf.py:323]:
```suggestion
```

File: docs/api.md [hunk 6]
```diff
@@ -169,4 +169,18 @@ RALPH_SENTRY_LRS_TRACES_SAMPLE_RATE=0.3
...
+
+Queries by `agent` or `authority` require pre-processing of the query 
+parameters. This pre-processing may be cached to avoid unnecessary 
+computing. You can set the maximum number of entries in your cache, which 
+should be greater than the number of unique agents queried for in a 
+sequence of requests. Defaults to 100. 
+
+```
+RALPH_QUERY_AGENTS_CACHE_MAX_SIZE=100
+```
```
Comment (wilbrdt) [inline docs/api.md:186]:
```suggestion
```

**Review by wilbrdt — COMMENTED** [review]
Some reminiscences of caching, but otherwise looks good to me!

**Pull request merged**